kratos/includes/properties.h

```diff
@@ -278,31 +293,303 @@ class Properties : public IndexedObject
+     * @return The pointer to the subproperty of interest
+     */
+    Properties::Pointer pGetSubPropertiesById(const IndexType SubPropertyIndex) const
+    {
+        // Looking into the database
+        const auto property_iterator = mSubPropertiesList.find(SubPropertyIndex);
+        if (property_iterator != mSubPropertiesList.end()) {
+            return *(property_iterator.base());
+        } else {
+            KRATOS_ERROR << "Subproperty ID: " << SubPropertyIndex << " is not defined on the current Properties ID: " << this->Id() << " creating a new one with ID: " << SubPropertyIndex << std::endl;
+            return nullptr;
+        }
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubPropertiesById(const IndexType SubPropertyIndex)
+    {
+        // Looking into the database
+        auto property_iterator = mSubPropertiesList.find(SubPropertyIndex);
+        if (property_iterator != mSubPropertiesList.end()) {
+            return *(property_iterator);
+        } else {
+            KRATOS_ERROR << "Subproperty ID: " << SubPropertyIndex << " is not defined on the current Properties ID: " << this->Id() << " creating a new one with ID: " << SubPropertyIndex << std::endl;
+            return *this;
+        }
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id (constant version)
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubPropertiesById(const IndexType SubPropertyIndex) const
+    {
+        // Looking into the database
+        if (mSubPropertiesList.find(SubPropertyIndex) != mSubPropertiesList.end()) {
+            return *(mSubPropertiesList.find(SubPropertyIndex));
+        } else {
+            KRATOS_ERROR << "Subproperty ID: " << SubPropertyIndex << " is not defined on the current Properties ID: " << this->Id() << std::endl;
+        }
+    }
+
+    /**
+     * @brief This method checks if the subproperty exists from the adress corresponding to the proper structure of indexes
+     * @param rAdress The text that indicates the structure of subproperties to iterate and found the property of interest
+     * @return True if there is such subproperty, false otherwise
+     */
+    bool HasSubPropertiesByAddress(const std::string& rAdress)
+    {
+        const auto indexes = TrimComponentName(rAdress);
+        if (indexes[0] == this->Id()) {
+            Properties::Pointer p_property_interest = Kratos::make_shared<Properties>(*this);
+            for (IndexType i_index = 1; i_index < indexes.size(); ++i_index) {
+                if ( !p_property_interest->HasSubPropertiesById(indexes[i_index])) {
```

Comment:
This implementation can give wrong results. We want to check if the subproperties is correspond to address which reflects also its position in hierarchy 

```
P1
-  P2
-  P3
    - P2
    - P4
```
Considering the above hierarchy the 1.2 and 1.3.2 are correct address but 1.2.3 or 3.1.1 should give false.

Reply:
True, I am so dummy sometimes

Reply:
Solved, I added https://github.com/KratosMultiphysics/Kratos/pull/2919/commits/980357f31d97daaf91f1df13f9604773d95ddd2f to check this behavior